src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1432,6 +1433,20 @@ private function registerPropertyInfoConfiguration(ContainerBuilder $container,
             $definition->addTag('property_info.description_extractor', ['priority' => -1000]);
             $definition->addTag('property_info.type_extractor', ['priority' => -1001]);
         }
+
+        if (!$container->getParameter('kernel.debug')) {
```

Comment:
IIRC, it's preferred to use XML for the service definition and to remove it in the extension is appropriate, to allow IDEs like PHPStorm to detect the services. Not sure if it's really matter here has it's the same interface that is implemented, it's more for consistency.

Reply:
I've followed what I've been done for the cache of ClassMetadataFactory:
https://github.com/symfony/symfony/blob/e19db439d86bce4dd5f6e328cc435702b12e7a42/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1508-L1520
If I change the code for this PR, I think it should be good to modify it for this part too.

Reply:
Let's do this, but this should target 4.3.